fix: remove username matching

janniks opened the pull request:
## scope
**EDIT:** now removes the `username` field completely from `auth`

~removes username matching from auth package (application-side)
currently — in applications using @stacks/connect — the @stacks/auth package will "validate" usernames which the wallet returned in the authResponse. as "validation" these usernames are matched to the response of the API's `/names` endpoint.~

~the matching is removed from [`verifyAuthResponse`](https://github.com/hirosystems/stacks.js/pull/1230/files#diff-916fca8273bf9c9b085f7a99720e76d1d5a974da0b1b237b4d9ff2d9686b48fe). in addition, the previously used `doPublicKeysMatchUsername` [seems incorrect](https://github.com/hirosystems/stacks-wallet-web/issues/2354#issuecomment-1093026523) and is removed completely (only used in `verifyAuthResponse` previously).~

## comments
~a second PR could investigate removing the `username` field from auth responses in general, as brought up [here](https://github.com/hirosystems/stacks.js/issues/1144#issuecomment-1099968763) (should be coordinated w/ wallet-web).~

janniks committed:
fix: remove username checking

janniks committed:
fix: remove doPublicKeysMatchUsername

kyranjamie reviewed (approved):
Nice improvement @janniks. 

This never made sense to me: if the registrar is down, you can't verify an auth response? 🤷🏼

kantai reviewed (approved):
This is fine to me, so long as you're all clear on what the goal is here: removing application validation of BNS names. Without this validation, applications can no longer trust that the BNS name that the authentication response provides is actually owned by the claimed user.

janniks committed:
feat: remove username from payload and userdata

BREAKING CHANGE: This change REMOVES the `username` attribute from auth token payloads and therefore also from userData in @stacks/connect. Hence, there is NO MORE username verification done by @stacks/connect automatically.

janniks committed:
docs: fix docstring

janniks committed:
chore: bump payload version
